crates/rspack_core/src/ensure_chunk_conditions_plugin.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashSet;
+
+use rspack_error::Error::InternalError;
+use rspack_identifier::Identifiable;
+
+use crate::{OptimizeChunksArgs, Plugin, PluginContext, PluginOptimizeChunksOutput};
+
+#[derive(Debug)]
+struct EnsureChunkConditionsPlugin;
```

Comment:
It would be better to move this plugin into a new crate.

Reply:
ok